code/go/internal/validator/semantic/validate_visualizations_used_by_value.go

```diff
@@ -0,0 +1,106 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package semantic
+
+import (
+	"fmt"
+	"log"
+	"path"
+
+	"github.com/pkg/errors"
+
+	ve "github.com/elastic/package-spec/code/go/internal/errors"
+	"github.com/elastic/package-spec/code/go/internal/fspath"
+	"github.com/elastic/package-spec/code/go/internal/pkgpath"
+)
+
+type reference struct {
+	ID   string
+	Name string
+	Type string
+}
+
+// ValidateVisualizationsUsedByValue warns if there are any Kibana
+// Dashboard that defines visualizations by reference instead of value.
+// That is, it warns if a Kibana dashbaord file, foo.json,
+// defines some visualization using reference (containing an element of
+// "visualization" type inside references key).
+func ValidateVisualizationsUsedByValue(fsys fspath.FS) ve.ValidationErrors {
+	var errs ve.ValidationErrors
+
+	filePaths := path.Join("kibana", "dashboard", "*.json")
+	objectFiles, err := pkgpath.Files(fsys, filePaths)
+	if err != nil {
+		errs = append(errs, errors.Wrap(err, "error finding Kibana Dashboard files"))
+		return errs
+	}
+
+	for _, objectFile := range objectFiles {
+		filePath := objectFile.Path()
+
+		objectReferences, err := objectFile.Values(`$.references`)
+		if err != nil {
+			// no references key in dashboard json
```

Comment:
There are test packages that do not have "references" key. To avoid errors in case of this key does not exist, this error is ignored.